In [146]:
import cv2
from skimage.transform import resize
from os import listdir, walk
from os.path import isfile, join
from matplotlib import pyplot as plt
import numpy as np
import pandas as pd
import random
from sklearn.model_selection import train_test_split
from tensorflow import keras
from tensorflow.keras import layers
from tensorflow.keras import backend as K

def getList(dict):
    """
    This function returns the list of keys in a dictionary
    """
    return dict.keys()

def image_reshape(directory, new_size = (300,300), gray =False):
    """
    give the list of photos with the correct pixel size you want to 
    downsize. If you want Gray then write True.
    """
    items_resized  =[]
    pictures = []
    labels = []
    mapping = {'Agaricus' : 0,'Amanita' : 1,'Boletus' : 2,'Cortinarius' : 3,'Entoloma' : 4,'Hygrocybe' : 5,
    'Lactarius' : 6,'Russula' : 7,'Suillus' : 8
                }
    list_of_mushrooms = getList(mapping)
    for f in list_of_mushrooms:
        pictures_dir = directory + f + '/'
        pictures = [_ for _ in listdir(pictures_dir) if isfile(join(pictures_dir , _))]
        for idx, picture in enumerate(pictures[:100]):
            item = cv2.imread(pictures_dir + picture)
            labels.append(mapping[f])
            
            if gray == True:
                item_gray = cv2.cvtColor(item, cv2.COLOR_BGR2GRAY)
                item_resized_gray = resize(item_gray, new_size)
                items_resized.append(item_resized_gray)
            else:
                item_resized_colored = resize(item, new_size)
                items_resized.append(item_resized_colored)
    return items_resized, labels

directory = 'data/Data/Mushrooms/' 
label , dir =  image_reshape(directory, new_size = (300,300))

Premature end of JPEG file


In [ ]:
def image_reshape(directory, new_size = (300,300), gray =False):
    """
    give the list of photos with the correct pixel size you want to 
    downsize. If you want Gray then write True.
    """

    label = []
    dir= []
    items_resized = []
    mapping = {'Agaricus' : 0,'Amanita' : 1,'Boletus' : 2,'Cortinarius' : 3,'Entoloma' : 4,'Hygrocybe' : 5,
    'Lactarius' : 6,'Russula' : 7,'Suillus' : 8
                }
    list_of_mushrooms = getList(mapping)
    for f in list_of_mushrooms:
        pictures_dir = directory + f + '/'
        pictures = [_ for _ in listdir(pictures_dir) if isfile(join(pictures_dir , _))]
        for idx, picture in enumerate(pictures):
            label.append(f)  
            dir.append(pictures_dir + picture) 
        
    return dir, label

In [147]:
def image_reshape(directory, new_size = (300,300), gray =False):
    """
    give the list of photos with the correct pixel size you want to 
    downsize. If you want Gray then write True.
    """

    label = []
    dir= []
    items_resized = []
    mapping = {'Agaricus' : 0,'Amanita' : 1,'Boletus' : 2,'Cortinarius' : 3,'Entoloma' : 4,'Hygrocybe' : 5,
    'Lactarius' : 6,'Russula' : 7,'Suillus' : 8
                }
    list_of_mushrooms = getList(mapping)
    for f in list_of_mushrooms:
        pictures_dir = directory + f + '/'
        pictures = [_ for _ in listdir(pictures_dir) if isfile(join(pictures_dir , _))]
        for idx, picture in enumerate(pictures):
            label.append(f)  
            dir.append(pictures_dir + picture) 
    for idx, dir in enumerate(dir):     
        item = cv2.imread(dir)
        if gray == True:
            item_gray = cv2.cvtColor(item, cv2.COLOR_BGR2GRAY)
            item_resized_gray = resize(item_gray, new_size)
            items_resized.append(item_resized_gray)
        else:
            item_resized_colored = resize(item, new_size)
            items_resized.append(item_resized_colored)
        
    return items_resized, label
    

directory = 'data/Data/Mushrooms/' 
label , dir =  image_reshape(directory, new_size = (300,300))

Premature end of JPEG file


In [148]:
print(label)
print(dir)

[array([[[0.86983987, 0.92474183, 0.93519935],
        [0.82414706, 0.88809804, 0.92806863],
        [0.82634967, 0.89045098, 0.9499281 ],
        ...,
        [0.03328105, 0.24615033, 0.18974183],
        [0.04479412, 0.2475098 , 0.19456863],
        [0.06756863, 0.23856209, 0.1989902 ]],

       [[0.87362418, 0.92903268, 0.94416667],
        [0.83847059, 0.90151961, 0.94465686],
        [0.84263072, 0.90472222, 0.96681373],
        ...,
        [0.03600327, 0.24646078, 0.18959804],
        [0.04570588, 0.25642157, 0.1905    ],
        [0.07180392, 0.26070261, 0.20060784]],

       [[0.85339869, 0.9121732 , 0.93813725],
        [0.84147059, 0.90617647, 0.95656863],
        [0.83866013, 0.90135621, 0.96707516],
        ...,
        [0.04531046, 0.25214052, 0.1995098 ],
        [0.03995098, 0.26651961, 0.18490196],
        [0.05727124, 0.2745098 , 0.19303922]],

       ...,

       [[0.00137255, 0.02120915, 0.03604575],
        [0.0245098 , 0.06598039, 0.08784314],
        [0.01593137, 

In [100]:
directory = 'data/Data/Mushrooms/' 
X , y = image_reshape(directory, new_size = (300,300))

In [105]:
type(y)

list

In [62]:
item = cv2.imread('data/Data/Mushrooms/Agaricus/000_ePQknW8cTp8.jpg') 


In [63]:
item

array([[[35, 36, 40],
        [37, 38, 42],
        [43, 44, 48],
        ...,
        [ 1,  3,  4],
        [ 9, 10, 14],
        [ 9, 10, 14]],

       [[21, 22, 26],
        [31, 32, 36],
        [33, 34, 38],
        ...,
        [ 1,  3,  4],
        [ 7,  8, 12],
        [ 7,  8, 12]],

       [[31, 32, 36],
        [42, 43, 47],
        [37, 38, 42],
        ...,
        [ 0,  1,  5],
        [ 4,  4, 10],
        [ 4,  4, 10]],

       ...,

       [[ 2,  4,  5],
        [ 0,  1,  2],
        [ 0,  1,  5],
        ...,
        [ 3,  0, 10],
        [ 3,  0, 11],
        [ 3,  0, 11]],

       [[ 4,  6,  7],
        [ 0,  1,  2],
        [ 0,  0,  3],
        ...,
        [ 3,  0, 10],
        [ 2,  0, 10],
        [ 2,  0, 10]],

       [[ 0,  1,  2],
        [ 0,  0,  1],
        [ 0,  0,  3],
        ...,
        [ 3,  0, 10],
        [ 5,  1, 13],
        [ 5,  1, 13]]], dtype=uint8)

In [109]:
test0 = pd.read_csv('data/Data/fungiclef2022/FungiCLEF2022_test_metadata.csv',sep = ',')

In [114]:
test0.head(3)

Unnamed: 0  ObservationId            eventDate  month   day countryCode  \
0           0     3305985310  2021-06-15T00:00:00    6.0  15.0          DK   
1           1     3305985310  2021-06-15T00:00:00    6.0  15.0          DK   
2           2     3305985310  2021-06-15T00:00:00    6.0  15.0          DK   

  Location_lvl0 Location_lvl3 Location_lvl2 Location_lvl1  \
0      Ellesøen       Denmark      Sjælland          Køge   
1      Ellesøen       Denmark      Sjælland          Køge   
2      Ellesøen       Denmark      Sjælland          Køge   

                    Substrate       Habitat          filename  
0  dead wood (including bark)  Bog woodland  0-3305985310.JPG  
1  dead wood (including bark)  Bog woodland  1-3305985310.JPG  
2  dead wood (including bark)  Bog woodland  2-3305985310.JPG

In [121]:
new_df = pd.DataFrame(['favorite_mushroom', '/hoz/go/'], columns=['label', 'dir'])
Table.append(new_df, ignore_index=True)

ValueError: Shape of passed values is (2, 1), indices imply (2, 2)

In [23]:
test1 = pd.read_csv('data/Data/fungiclef2022/DF20-train_metadata.csv',sep = ',')

In [32]:
test1.head(3)
test1.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 266344 entries, 0 to 266343
Data columns (total 34 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   gbifID                266344 non-null  int64  
 1   eventDate             266328 non-null  object 
 2   year                  266328 non-null  float64
 3   month                 266328 non-null  float64
 4   day                   266328 non-null  float64
 5   countryCode           266344 non-null  object 
 6   locality              266273 non-null  object 
 7   taxonID               266344 non-null  float64
 8   scientificName        266344 non-null  object 
 9   kingdom               266344 non-null  object 
 10  phylum                266344 non-null  object 
 11  class                 266315 non-null  object 
 12  order                 266187 non-null  object 
 13  family                263975 non-null  object 
 14  genus                 266344 non-null  object 
 15  

In [28]:
mash = list(test1.scientificName.unique())

In [41]:
test2 = pd.read_csv('data/Data/fungiclef2022/DF20-val_metadata.csv',sep = ',')
len(mash)

1604

In [36]:
test2.head(2)
test2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29594 entries, 0 to 29593
Data columns (total 34 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   gbifID                29594 non-null  int64  
 1   eventDate             29593 non-null  object 
 2   year                  29593 non-null  float64
 3   month                 29593 non-null  float64
 4   day                   29593 non-null  float64
 5   countryCode           29594 non-null  object 
 6   locality              29584 non-null  object 
 7   taxonID               29594 non-null  float64
 8   scientificName        29594 non-null  object 
 9   kingdom               29594 non-null  object 
 10  phylum                29594 non-null  object 
 11  class                 29591 non-null  object 
 12  order                 29577 non-null  object 
 13  family                29330 non-null  object 
 14  genus                 29594 non-null  object 
 15  specificEpithet    

In [40]:
mash = list(test2.scientificName.unique())
len(mash)

1604

- From Big Data mushrooms we have two lists, splitted in test and validation.
- - 1604 kinds of fungis
- - 266344 / 19594 , test/ val


- Smalll data set
- - Categories : 'Agaricus' : 0,'Amanita' : 1,'Boletus' : 2,'Cortinarius' : 3,'Entoloma' : 4,'Hygrocybe' : 5,
    'Lactarius' : 6,'Russula' : 7,'Suillus' : 8
- - number of pics per category: [352, 749, 1072, 835, 363, 315, 1562, 1147, 310]